s2spy/preprocess.py

```diff
@@ -88,12 +124,53 @@ def _check_input_data(data: Union[xr.DataArray, xr.Dataset]):
         )
 
 
+def _check_temporal_resolution(timescale: Timescale) -> Timescale:
+    support_temporal_resolution = ["monthly", "weekly", "daily"]
+    if timescale not in support_temporal_resolution:
+        raise ValueError(
+            "Given temporal resoltuion is not supported."
+            "Please choose from 'monthly', 'weekly', 'daily'."
+        )
+    return timescale
+
+
+def _check_data_resolution_match(
+    data: Union[xr.DataArray, xr.Dataset], timescale: Timescale
+):
+    """Check if the temporal resolution of input is the same as given timescale."""
+    timescale_dict = {
+        "monthly": np.timedelta64(1, "M"),
+        "weekly": np.timedelta64(1, "W"),
+        "daily": np.timedelta64(1, "D"),
+    }
+    time_intervals = np.diff(data["time"].to_numpy())
+    temporal_resolution = np.median(time_intervals).astype("timedelta64[D]")
+    if timescale == "monthly":
+        temporal_resolution = temporal_resolution.astype(int)
+        min_days, max_days = (28, 31)
+        if not max_days > temporal_resolution > min_days:
```

Comment:
Hi Yang, 

I pulled the branch to test it :) 

In my test, `temporal_resolution` for monthly data is 31. According to the statement, `max_days` (31) is not larger than 31 ;) 
`if not max_days > temporal_resolution > min_days:`

Should be:
`if not max_days >= temporal_resolution >= min_days:`

Reply:
Oh! Yes, of course! Thanks for fixing this!